### PR TITLE
[PT-BR] Translate Conditional Types and Unknown and Never

### PR DESCRIPTION
**Pull request by vinicius98s**

### CODE REVIEW

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+// Tipos Condicionais fornecem uma maneira simples de adicionar
+// lógica no sistema de tipos do Typescript. Esse é de fato um recurso
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// lógica no sistema de tipos do Typescript. Esse é um recurso
```

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+// avançado, e é bastante possível que você não precise utilizá-lo
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// avançado, e é muito possível que você não precise utilizá-lo
```

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+// no seu código do dia a dia.
+
+// Um tipo condicional se parece como:
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// Um tipo condicional se parece com:
```

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+//
+//   A extends B ? C : D
+//
+// Onde a condição é se um tipo se extende a uma expressão,
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// Onde a condição é: se um tipo se extende a uma expressão,
```

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+// e que tipo deveria ser retornado.
+
+// Vamos passar por alguns exemplos, por questões de brevidade
+// usaremos apenas uma letra para tipos genéricos. Isso é opcional
+// mas nos restringindo à 60 caracteres fica difícil caber na tela.
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// mas nos restringindo à 60 caracteres, pois fica difícil caber na tela.
```

Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// usaremos apenas uma letra para tipos genéricos. Isso é opcional,
// mas nos restringimos à 60 caracteres para caber na tela.
```

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+// Isso é chamado de tipo condicional distributivo porque
+// o tipo distribui para cada membro da união.
+
+// Tipos condicionais diferidos
+
+// Tipos condicionais podem ser usados para diminuir suas APIs
+// na qual podem retornar diferentes tipos dependendo dos inputs.
+
+// Por exemplo essa função na qual pode retornar tanto
+// uma string quanto um number dependendo do boolean passado.
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// Tipos condicionais podem ser usados para diminuir suas APIs
// que podem retornar diferentes tipos dependendo dos inputs.

// Por exemplo, essa função pode retornar tanto uma string
// quanto um number dependendo do boolean passado.
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 7]
```diff
@@ -0,0 +1,125 @@
+// Unknown
+
+// Unknown é um daqueles tipos que assim que é entendio,
+// você acha muitos casos de uso. Ele age como um irmão para o tipo any.
+// Onde any permite ambiguidade - unkown requer especifidades.
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
// Unknown é um daqueles tipos que assim que é entendido,
// você acha muitos casos de uso. Ele age como um irmão para o tipo any.
// any permite ambiguidade - unknown requer especificidades.
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+// O objeto minhaOutraConta não pode ser usado até o tipo ser declarado
+// para o TypeScript. Isso pode ser usado para garantir que
+// quem consumir a API pense em sua tipagem com antecedência:
+
+type Usuario = { nome: string };
+const minhaContaDeUsuario = transformadorJsonUnkown(`{ "nome": "Samuel" }`) as Usuario;
+minhaContaDeUsuario.nome;
+
+// Unknown é uma boa ferramenta, para entende-lo ainda mais leia:
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
// Unknown é uma boa ferramenta, para entendê-lo ainda mais leia:
```

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+type Cachorro = { latido: true };
+type Guepardo = { miau: true; rapido: true };
+type Lobo = { latido: true; uivos: true };
+
+// Podemos criar um tipo condicional onde é possível extrair
+// tipos que se aplicam apenas com algo que late.
+
+type ExtrairLatidos<A> = A extends { latido: true } ? A : never;
+
+// Assim podemos criar tipos envolvidos pelo ExtrairLatido:
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// Assim podemos criar tipos envolvidos pelo ExtrairLatidos:
```

**Review by danilofuchs — CHANGES_REQUESTED** [review]
Great job!
I have a few suggestions and caught some typos

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+// você irá receber um tipo de retorno diferente:
+let retornoDeString = pegarID(true);
+let retornoDeNumber = pegarID(false);
+let stringOuNumber = pegarID(Math.random() < 0.5);
+
+// Nesse caso acima o TypeScript sabe o tipo de retorno imediatamente.
+// Contudo, você pode usar tipos condicionais em funções
+// onde o tipo não é conhecido. Isso é chamado tipo condicional diferido.
+
+// O mesmo que o nosso ExtrairLatido acima, mas como uma função
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// O mesmo que o nosso ExtrairLatidos acima, mas como uma função
```

File: packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+declare function extrairMiado<T>(x: T): T extends { miau: true } ? T : undefined;
+
+// Existe uma ferramenta extra útil dentro dos tipos condicionais, na qual
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Meta-Types/Conditional Types.ts]:
```suggestion
// Existe uma ferramenta muito útil dentro dos tipos condicionais, na qual
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 12]
```diff
@@ -0,0 +1,125 @@
+// Unknown
+
+// Unknown é um daqueles tipos que assim que é entendio,
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
// Unknown é um daqueles tipos que, assim que você entende,
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+const minhaContaDeUsuario = transformadorJsonUnkown(`{ "nome": "Samuel" }`) as Usuario;
+minhaContaDeUsuario.nome;
+
+// Unknown é uma boa ferramenta, para entende-lo ainda mais leia:
+// https://mariusschulz.com/blog/the-unknown-type-in-typescript
+// https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type
+
+// Never
+
+// Porque o Typescript suporta análise de fluxo do código, a linguagem
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
// Como o Typescript suporta análise de fluxo do código, a linguagem
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+};
+
+// Se você passar o mouse em cima da função, verá () => never
+// que significa que isso nunca deverá acontecer. Estes ainda
+// podem ser passados como outros valores:
+
+const meuValor = nuncaRetorna();
+
+// Tendo uma função que retorna never pode ser útil quando lidamos com
+// a imprevisiblidade do tempo de execução do JavaScript e os
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
// a imprevisiblidade do runtime JavaScript e os
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  // De acordo com o sistema de tipos, esse caminho do código
+  // nunca deveria acontecer, que combina com o tipo de
+  // retorno do nuncaRetorna.
+
+  return nuncaRetorna();
+};
+
+// O estado da definição de tipos indicam que um usuário deve ser
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
// O estado da definição de tipos indica que um usuário deve ser
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+};
+
+// Você vai ter um erro de compilação falando que seu
+// novo tipo de flor não pode ser convertido em never.
+
+// Never em Unions
+
+// O never é removido automaticamente do tipo union.
+
+type NeverÉRemovido = string | never | number;
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
type NeverERemovido = string | never | number;
```

File: packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+
+// Se você olhar no tipo para NeverÉRemovido, verá que
```
Comment (khaosdoctor) [inline packages/playground-examples/copy/pt/TypeScript/Primitives/Unknown and Never.ts]:
```suggestion
// Se você olhar no tipo para NeverERemovido, verá que
```

**Review by khaosdoctor — CHANGES_REQUESTED** [review]
Small changes

**Comment by khaosdoctor** [comment]
@vinicius98s Corrected a few examples!

**Comment by vinicius98s** [comment]
@khaosdoctor @danilofuchs Applied those suggestions and corrected the typos

**Review by danilofuchs — APPROVED** [review]
LGTM

**Comment by danilofuchs** [comment]
@orta the merge action failed mysteriously 🤔

**Comment by khaosdoctor** [comment]
lgtm

**Comment by khaosdoctor** [comment]
@danilofuchs Well... I don't know what happened

**Comment by orta** [comment]
I know, I had hoped this would solve it https://github.com/orta/code-owner-self-merge/issues/6

**Comment by orta** [comment]
Hrm, 6 hours ago might have been before I shipped the update - so I'm not sold on it _not_ being fixed actually - just remember to leave a review LGTM next time and we'll see

**Pull request merged**